### PR TITLE
Blog:  Building out UNS with FlowFuse

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+---
+title: Building out uns with FlowFuse
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:2]:
```suggestion
title: Building a Unified Namespace (UNS) with FlowFuse
```

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+subtitle: FlowFuse provides a complete toolkit for implementing a Unified Namespace seamlessly
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:3]: Implement your Unified Namespace seamlessly using our low-code platform

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+   - flowfuse
+   - uns
+   - mqtt
+---
+
+As operations become more connected, managing data from different systems and devices can be challenging. A Unified Namespace (UNS) centralizes your data, making it easy to access and use across your network. This article demonstrates how to build a UNS with FlowFuse, streamlining data flow and enabling real-time insights.
+
+<!--more-->
+
+## **What is the Unified Namespace?**
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:18]: We can skip this part and link UNS is the paragraph above.

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+![Unified Namespace](./images/uns.png){data-zoomable}
+_Unified Namespace_
+
+A Unified Namespace (UNS) is a data architecture that allows connecting all your systems, sensors, and machines into a single, easy-to-access network. It helps organize data from different sources consistently, making it simple for everyone in your business to share and find the information they need. By unifying your data, UNS speeds up operations, simplifies system integration, and provides faster insights to improve productivity.  
+
+For more information on UNS, please read our article [What is UNS](/blog/2023/12/introduction-to-unified-namespace/).
+
+## **The Role of FlowFuse in Industrial Data Operations and Building Your UNS**
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:27]: This section can be merged into the intro paragraph above.

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+[FlowFuse](/) bridges the gap between legacy systems and modern technologies. It enables seamless data collection from edge devices, systems, and services while allowing business applications to contribute data to the UNS through its data pipelines.
+
+FlowFuse offers a complete toolkit for building and managing your UNS, featuring Node-RED for data orchestration, MQTT for real-time communication, and an enterprise layer for centralized management and collaboration. This unified solution simplifies the integration of devices, applications, and legacy systems, ensuring smooth data flow and real-time insights across your organization.
+
+## **Building UNS with FlowFuse**
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:33]:
```suggestion
## **Building a UNS with Real-Time Sensor Data**
```

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+In this section, we'll walk you through building a UNS with FlowFuse using a practical example. We’ll send real-time data from a Raspberry Pi connected to an ADXL345 sensor to the UNS.
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:35]:
```suggestion
This example demonstrates how to build a UNS using FlowFuse and a Raspberry Pi with an ADXL345 sensor. We'll cover data acquisition, processing, formatting, and publishing to the UNS, highlighting the benefits of standardized data management and real-time insights.
```

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+As mentioned earlier, Node-RED supports various industrial protocols, from legacy to modern ones. While older protocols like Modbus and OPC-UA were initially designed for machine-to-machine (M2M) communication and cannot directly interact with cloud systems, Node-RED bridges this gap. Using Node-RED, you can collect data from these legacy systems, transform and format it using low-code workflows, and then send it to the cloud via protocols like MQTT, Kafka, AMQP, or others.
+
+In our example, Node-RED can directly collect metrics from sensors using [I2C](https://flows.nodered.org/node/node-red-contrib-i2c) on the Raspberry Pi. This approach simplifies the process by eliminating the need for additional communication layers. To run Node-RED on the Raspberry Pi, we use [FlowFuse Device Agent](/product/device-agent/), which enables you to remotely monitor and build flows securely from the FlowFuse platform in your edge devices. For more details on how to set up and run FlowFuse Device Agent on different devices, refer to this [documentation section](/node-red/hardware/).
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:45]:
```suggestion
In our example, Node-RED can directly collect metrics from sensors using [I2C](https://flows.nodered.org/node/node-red-contrib-i2c) on the Raspberry Pi. This approach simplifies the process by eliminating the need for additional communication layers. To run Node-RED on the Raspberry Pi, we use [FlowFuse Device Agent](/product/device-agent/), which enables you to remotely monitor and build flows securely from the FlowFuse platform in your edge devices. [See here](/node-red/hardware/) for more details on how to set up and run FlowFuse Device Agent on different devices. 
```

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+### **Step 2: Transform and Process the Collected metrics**
+
+Once you've collected data from your devices, the next step is to transform it using Node-RED. Industrial systems often use different protocols (e.g., Modbus, OPC UA), and each might have its own data structure, which can create challenges for integration—for example, the ADXL345 sensor outputs raw data as electrical signals (buffer data). We first need to format it into a human-readable format and then calculate the Magnitude, a standard vibration monitoring unit.
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:49]:
```suggestion
Once you've collected data from your devices, the next step is to transform it using Node-RED. Industrial systems often use different protocols such as Modbus and OPC UA, and each might have its own data structure, which can create challenges for integration.For example, the ADXL345 sensor outputs raw data as electrical signals (buffer data). We first need to format it into a human-readable format and then calculate the Magnitude, a standard vibration monitoring unit.
```

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+For information on how to use MQTT with Node-RED, refer to [Using MQTT with Node-RED](/blog/2024/06/how-to-use-mqtt-in-node-red/)
+
+After deploying, you can monitor the topic hierarchy on the FlowFuse platform by switching to the "Hierarchy" tab in the Broker interface.
+
+![Monitoring your mqtt topic hierarchy within FlowFuse](./images/topic-hierarchy.png){data-zoomable}
+_Monitoring your mqtt topic hierarchy within FlowFuse_
+
+Once your data is in the UNS, you have a centralized, real-time view of your operations. This unified structure enables easier access, sharing, and analysis of data across systems, helping you drive better decisions, improve efficiency, and gain valuable insights to optimize your processes.
+
+With real-time data access, you can create monitoring dashboards using the FlowFuse Dashboard with a low-code approach, integrate with other cloud solutions, or leverage it further for enhanced analytics or automation.
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:195]:
```suggestion
With real-time data access, you can create monitoring dashboards using the [FlowFuse Dashboard](https://flowfuse.com/product/dashboard/) with a low-code approach, integrate with other cloud solutions, or leverage it further for enhanced analytics or automation.
```

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+
+![FlowFuse Dashboard Monitoring Vibrations](./images/flowfuse-dashboard.png){data-zoomable}
+_FlowFuse Dashboard Monitoring Vibrations_
+
+## **Final Thoughts**
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:200]:
```suggestion
## **Conclusion**
```

Comment (Hasmin-AC) [reply src/blog/2024/11/building-uns-with-flowfuse.md:200]: Final thoughts indicate new points being presented.

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+This section explains how to set up a Unified Namespace (UNS) using FlowFuse, a Raspberry Pi, and an ADXL345 sensor. The Raspberry Pi collects data from the sensor, which we collect and process in Node-RED, calculate vibration magnitude, format it, and send it to the UNS using standardized topic names.
+
+### **Step 1: Collect Metrics from Devices**
+
+The first step in building your UNS is collecting data from your devices. The method you choose will depend on the type of device and the communication protocol it supports. For instance, many devices use serial communication standards like Modbus, while others, such as industrial controllers, might use OPC-UA.
+
+As mentioned earlier, Node-RED supports various industrial protocols, from legacy to modern ones. While older protocols like Modbus and OPC-UA were initially designed for machine-to-machine (M2M) communication and cannot directly interact with cloud systems, Node-RED bridges this gap. Using Node-RED, you can collect data from these legacy systems, transform and format it using low-code workflows, and then send it to the cloud via protocols like MQTT, Kafka, AMQP, or others.
```
Comment (Hasmin-AC) [inline src/blog/2024/11/building-uns-with-flowfuse.md:31]:
```suggestion
Node-RED supports various industrial protocols, from legacy to modern ones. While older protocols like Modbus and OPC-UA were initially designed for machine-to-machine (M2M) communication and cannot directly interact with cloud systems, Node-RED bridges this gap. Using Node-RED, you can collect data from these legacy systems, transform and format it using low-code workflows, and then send it to the cloud via protocols like MQTT, Kafka, AMQP, or others.
```

**Review by Hasmin-AC — APPROVED** [review]

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+<!--more-->
+
+[FlowFuse](/) makes building a UNS simple. It connects old and new systems, collects data from devices and applications, and streamlines workflows. With tools like Node-RED for data flow, MQTT for real-time updates, and a central management layer, FlowFuse helps you improve efficiency and make better decisions.
+
+This article will show you how to build your UNS using FlowFuse, step by step.
+
+## **Building a UNS with Real-Time Sensor Data**
+
+This section explains how to set up a Unified Namespace (UNS) using FlowFuse, a Raspberry Pi, and an ADXL345 sensor. The Raspberry Pi collects data from the sensor, which we collect and process in Node-RED, calculate vibration magnitude, format it, and send it to the UNS using standardized topic names.
+
```
Comment (ZJvandeWeg) [inline src/blog/2024/11/building-uns-with-flowfuse.md:26]: If we're building an UNS, shouldn't we also do something with the data? Not just publish it?

File: src/blog/2024/11/building-uns-with-flowfuse.md [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+#### **Example: Transforming Data from ADXL345 Sensor (Raspberry Pi)**
+
+Raw data from the ADXL345 sensor might look like this:
+
+```json 
+[26,0,244,255,37,255]  
+```  
+
+Using a [Function node](/node-red/core-nodes/function/) in Node-RED, we can convert this into a human-readable format.
```
Comment (ZJvandeWeg) [inline src/blog/2024/11/building-uns-with-flowfuse.md:55]: Why not a change node?

**Pull request merged**